### ISSUE TITLE
Attach sonar analysis results to new PRs

### ISSUE THREAD

**Issue by przemyslawjacewicz** [openaire/iis#1024]
The goal of this issue is to attach sonar analysis results to new pull requests in IIS.

**Comment by przemyslawjacewicz** [comment]
I've done some research and it seems that with our current setup it's not possible to attach the results of Sonar analysis to PRs.

The reason is that our version of Sonar does not allow to decorate PRs with analysis results. We're running Community version, which is free and decorating PRs is supported starting with Developer version, which is not free. So we would have to upgrade our Sonar setup.

**Comment by przemyslawjacewicz** [comment]
The second thing is that our current setup does not work properly - Sonar results do not reflect the current version of master branch. The reason for this is probably due to changes in Jenkins build configuration. Currently we run Sonar analysis in `iis_build_multibranch` Jenkins build. This configuration builds and runs Sonar analysis for each branch. As a result Sonar shows the results of the last branch build and analyzed. I think that within out current setup the best thing we can do is to run the analysis on master branch only, excluding the analysis of other branches. This would require this type of a workaround in current Jenkins configuration:
```bash
        stage("Package") {
            steps {
                configFileProvider([configFile(fileId: "iis-build-properties", variable: 'BUILD_PROPERTIES')]) {
                    load "${BUILD_PROPERTIES}"
                    withEnv(["JAVA_HOME=${ tool type: 'jdk', name: "$JDK_VERSION" }",
                             "PATH+MAVEN=${tool type: 'maven', name: "$MAVEN_VERSION"}/bin:${env.JAVA_HOME}/bin"]) {
                        withSonarQubeEnv('sonar.ceon.pl') {
                            sh '''
                                if [ $GIT_BRANCH = "master" ]; then
                                    mvn clean package \
                                        -DskipITs \
                                        -Djava.net.preferIPv4Stack=true \
                                        $SONAR_MAVEN_GOAL \
                                        -Dsonar.host.url=$SONAR_HOST_URL
                                else
                                    mvn clean package \
                                        -DskipITs \
                                        -Djava.net.preferIPv4Stack=true
                                fi
                            '''
                        }
                    }
                }
            }
        }
```
This is not a very good workflow because the analysis is run after merging into master. So if a problem in code is introduced in a PR then we can find only after merging and to fix it we would have to issue another PR. But it is still better then no analysis at all.
